src/frontend/components/QuillEditor/EditorToolbar.jsx

```diff
@@ -118,69 +118,71 @@ const formats = [
 ];
 
 // Quill Toolbar component
-const QuillToolbar = () => (
-  <div id="toolbar">
-    <span className="ql-formats">
-      <select className="ql-font" defaultValue="sans-serif">
-        <option value="sans-serif">Sans Serif</option>
-        <option value="serif">Serif</option>
-        <option value="monospace">Monospace</option>
-      </select>
-      <select className="ql-size" defaultValue="normal">
-        <option value="small">Small</option>
-        <option value="normal">Normal</option>
-        <option value="large">Large</option>
-        <option value="huge">Huge</option>
-      </select>
-      <select className="ql-header" defaultValue="3">
-        <option value="1">Heading</option>
-        <option value="2">Subheading</option>
-        <option value="3">Normal Text</option>
-      </select>
-    </span>
-    <span className="ql-formats">
-      <button className="ql-bold" />
-      <button className="ql-italic" />
-      <button className="ql-underline" />
-      <button className="ql-strike" />
-    </span>
-    <span className="ql-formats">
-      <button className="ql-list" value="ordered" />
-      <button className="ql-list" value="bullet" />
-      <button className="ql-indent" value="-1" />
-      <button className="ql-indent" value="+1" />
-    </span>
-    <span className="ql-formats">
-      <button className="ql-script" value="super" />
-      <button className="ql-script" value="sub" />
-      <button className="ql-blockquote" />
-      <button className="ql-direction" value="rtl" />
-    </span>
-    <span className="ql-formats">
-      <select className="ql-align" />
-      <select className="ql-color" />
-      <select className="ql-background" />
-    </span>
-    <span className="ql-formats">
-      <button className="ql-link" />
-      <button className="ql-image" />
-      <button className="ql-video" />
-    </span>
-    <span className="ql-formats">
-      <button className="ql-formula" />
-      {/* <button className="ql-code" /> */}
-      <button className="ql-code-block" />
-      <button className="ql-clean" />
-    </span>
-    <span className="ql-formats">
-      <button className="ql-undo">
-        <UndoButtonIcon />
-      </button>
-      <button className="ql-redo">
-        <RedoButtonIcon />
-      </button>
-    </span>
-  </div>
-);
+const QuillToolbar = () => {
+  return (
+    <div id="toolbar">
```

Comment:
I see that the below code is redundant. After you get the bandwidth (maybe the submission), you could try mapping through and then rendering it.

Reply:
yes I need to understand react-quill in detail